infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -230,9 +230,18 @@ private String getConflictHashCode(UUID streamId, byte[] conflictParam) {
     private TokenType txnCanCommit(TxResolutionInfo txInfo, /** Input. */
                                   AtomicReference<byte[]> conflictKey /** Output. */) {
         log.trace("Commit-req[{}]", txInfo);
-        final long txSnapshotTimestamp = txInfo.getSnapshotTimestamp();
+        final Token txSnapshotTimestamp = txInfo.getSnapshotTimestamp();
+
+        // A transaction can start with a timestamp issued from a previous
+        // epoch, so we need to reject transactions that don't have a snapshot
```

Comment:
Also, I think there is an extra 'don't'.. the logic is rejecting transactions that have a timestamp with a different epoch than the sequencer.

Reply:
I'll change it.